scrapy/http/request/__init__.py

```diff
@@ -78,6 +95,15 @@ def __init__(
         self._cb_kwargs = dict(cb_kwargs) if cb_kwargs else None
         self.flags = [] if flags is None else list(flags)
 
+    def _set_xback(self, name: str, value: Optional[Callable]) -> None:
+        if (
+            value is not None
+            and (name != "callback" or value is not NO_CALLBACK)
+            and not callable(value)
+        ):
```

Comment:
I wonder if the below is slightly easier to understand:

```suggestion
        if not (
            value is None
            or callable(value)
            or (name == "callback" and value is NO_CALLBACK)
        ):
```

Feel free to ignore.